sdk/python/kfp/components/_components.py

```diff
@@ -457,6 +467,26 @@ def expand_command_part(arg) -> Union[str, List[str], None]:
 
             return output_filename
 
+        elif isinstance(arg, InputUriPlaceholder):
+            input_name = arg.input_name
+            input_argument = argument_values.get(input_name, None)
+            if input_name in argument_values:
+                input_uri = input_uri_generator(input_name)
+                input_uris[input_name] = input_uri
+                return input_uri
+            else:
+                input_spec = inputs_dict[input_name]
+                if input_spec.optional:
+                    return None
+                else:
+                    raise ValueError('No value provided for input {}'.format(input_name))
+
+        elif isinstance(arg, OutputUriPlaceholder):
+            output_name = arg.output_name
+            output_uri = output_uri_generator(output_name)
+            output_uris[output_name] = output_uri
```

Comment:
Shall we check for conflicting path here, just like `OutputPathPlaceholder`?

Reply:
Done. I'll make a separate PR to do the same in v2.

Reply:
Actually, we have the check already in v2 code.